packages/frontend-shared/src/gql-components/CloudViewerAndProject.vue

```diff
@@ -0,0 +1,133 @@
+<script setup lang="ts">
```

Comment:
Should this be a composable? This component just triggers side-effects and has no template.

Reply:
I don't really see a difference between queries and mutations, and mutations are allowed outside of components. Maybe the pattern I'm more familiar with is:

```html
<CloudViewerAndProject>
  <App />
</CloudViewerAndProject>
```

The example above, it's more apparent that `CloudViewerAndProject` is having an effect on the App whereas our use case it seems like you could remove it (though that would cause everything to break).

I also stumbled upon [setup-stores](https://pinia.vuejs.org/core-concepts/#setup-stores) which could be interesting.

Not going to block on this, but a detailed comment about what this component is doing would be useful since figuring out why these queries are being executed is indirect.

Reply:
Nice, I will add a comment